libvast/vast/data.hpp

```diff
@@ -356,6 +357,48 @@ struct sum_type_access<vast::data> : default_sum_type_access<vast::data> {};
 
 } // namespace caf
 
+namespace vast {
+
+// -- extraction -------------------------------------------------------
+
+/// Tries to retrieve the value associated to `name` from `xs`.
+const data* get_if(const record* xs, std::string_view name);
+
+/// Tries to retrieve the value associated to `name` from `xs`.
+template <class T>
+caf::expected<T> get_if(const record* xs, std::string_view name) {
```

Comment:
```suggestion
template <class T>
const T* get_if(const record* xs, std::string_view name)
```

This interface seems to be the analogue of the overload above. Why do you induce a copy here when a pointer works as well?

Also, is it really that important that the caller get's an `error` back? A `nullptr` can mean either not a `T` or not a valid `name`, which seems fine at first.

Reply:
> This interface seems to be the analogue of the overload above. Why do you induce a copy here when a pointer works as well?

I don't have a strong opinion on this.

> Also, is it really that important that the caller get's an error back? A nullptr can mean either not a T or not a valid name, which seems fine at first.

The function observes the information about what exactly went wrong during its evaluation. Since it has multiple points of failure, I want it to say which one it was. If you are not interested in that error when calling the function, you can easily decay the returned value to a pointer.